storage/remote/codec.go

```diff
@@ -625,9 +625,11 @@ func exemplarProtoToExemplar(ep prompb.Exemplar) exemplar.Exemplar {
 }
 
 // HistogramProtoToHistogram extracts a (normal integer) Histogram from the
-// provided proto message. The caller has to make sure that the proto message
-// represents an integer histogram and not a float histogram.
+// provided proto message.
 func HistogramProtoToHistogram(hp prompb.Histogram) *histogram.Histogram {
+	if hp.IsFloatHistogram() {
+		panic("don't call HistogramProtoToHistogram on a float histogram")
```

Comment:
Minor nit: I think we usually state what went wrong in the panic. In this case, that would be more like "HistogramProtoToHistogram called with a float histogram".